Utils/azuretools-core/src/com/microsoft/azure/toolkit/lib/common/operation/AzureOperationRef.java

```diff
@@ -24,25 +24,30 @@
 
 import lombok.Builder;
 import lombok.Getter;
-import lombok.Setter;
 
 import java.lang.reflect.Method;
+import java.util.Objects;
 
 @Getter
-@Setter
 @Builder
 public class AzureOperationRef {
-    private Method method;
-    private String[] paramNames;
-    private Object[] paramValues;
-    private Object instance;
+    private final Method method;
+    private final String[] paramNames;
+    private final Object[] paramValues;
+    private final Object instance;
 
     @Override
     public boolean equals(final Object obj) {
         if (!(obj instanceof AzureOperationRef)) {
             return false;
         }
         final AzureOperationRef operation = (AzureOperationRef) obj;
-        return operation.getMethod() == this.method || operation.getMethod().equals(this.method);
+        return operation.getMethod() == this.method || Objects.equals(operation.getMethod(), this.getMethod());
+    }
+
+    @Override
+    public String toString() {
+        final AzureOperation annotation = AzureOperationUtils.getAnnotation(this);
+        return String.format("{title:'%s', method:%s}", annotation.value(), method.getName());
     }
```

Comment:
do we need to check whether annotation == null here?

Reply:
no, the annotation will never be null, otherwise this `AzureOperationRef` instance will not be created.